feat(rollback mutations): Recompile 100 times

derrickstolee opened the pull request:
The CompilingProcess has a hard-coded three retries to fix any
compiler errors that are fixed by recompiling. Instead, loop until
the recompile works, or until the recompile fails 100 times in a row.

Thanks to @Mobrockers for the assistance in with the variable
names to properly run the recompile.

Resolves one issue for #462.

Mobrockers commented:
I fixed the one failing unit test for you and added some logging so we can diagnose how many recompilations we use. Hope you don't mind.

derrickstolee commented:
> I fixed the one failing unit test for you and added some logging so we can diagnose how many recompilations we use. Hope you don't mind.

I absolutely don't mind. I remember seeing the unit test that was associated when I first made my change and completely forgot about it when changing the infinite loop to the recommended finite loop. Thanks for all the help here!
